src/protocols/block_io.rs

```diff
@@ -0,0 +1,67 @@
+//! Block I/O Protocol
+//!
+//! Used to abstract mass storage devices to allow code running in the EFI boot services environment
+//! to access the storage devices without specific knowledge of the type of device or controller that
+//! manages the device.
+
+pub const PROTOCOL_GUID: crate::base::Guid = crate::base::Guid::from_fields(
+    0x964e5b21,
+    0x6459,
+    0x11d2,
+    0x8e,
+    0x39,
+    &[0x0, 0xa0, 0xc9, 0x69, 0x72, 0x3b],
+);
+
+pub const REVISION: u64 = 0x0000000000010000u64;
+pub const REVISION2: u64 = 0x0000000000020001u64;
+pub const REVISION3: u64 = 0x000000000002001fu64;
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub struct BlockIoMedia {
```

Comment:
Similar to the disk-io protocols, I would drop the `BlockIo` prefix here, as it is already part of the module-name.